docs/serving/samples/hello-world/helloworld-java-spring/index.md

```diff
@@ -5,4 +5,249 @@ weight: 1
+   [Spring Initializr](https://start.spring.io/) page. Specify Artifact as
+   `helloworld` and add the `Web` dependency. Then click `Generate Project`,
+   download and unzip the sample archive.
+
+1. Update the `SpringBootApplication` class in
+   `src/main/java/com/example/helloworld/HelloworldApplication.java` by adding a
+   `@RestController` to handle the "/" mapping and also add a `@Value` field to
+   provide the TARGET environment variable:
+
+   ```java
+   package com.example.helloworld;
+
+   import org.springframework.beans.factory.annotation.Value;
+   import org.springframework.boot.SpringApplication;
+   import org.springframework.boot.autoconfigure.SpringBootApplication;
+   import org.springframework.web.bind.annotation.GetMapping;
+   import org.springframework.web.bind.annotation.RestController;
+
+   @SpringBootApplication
+   public class HelloworldApplication {
+
+     @Value("${TARGET:World}")
+     String target;
+
+     @RestController
+     class HelloworldController {
+       @GetMapping("/")
+       String hello() {
+         return "Hello " + target + "!";
+       }
+     }
+
+     public static void main(String[] args) {
+       SpringApplication.run(HelloworldApplication.class, args);
+     }
+   }
+   ```
+
+1. Run the application locally:
+
+   ```shell
+   ./mvnw package && java -jar target/helloworld-0.0.1-SNAPSHOT.jar
+   ```
+
+   Go to `http://localhost:8080/` to see your `Hello World!` message.
+
+1. In your project directory, create a file named `Dockerfile` and copy the code
+   block below into it. For detailed instructions on dockerizing a Spring Boot
+   app, see
+   [Spring Boot with Docker](https://spring.io/guides/gs/spring-boot-docker/).
+   For additional information on multi-stage docker builds for Java see
+   [Creating Smaller Java Image using Docker Multi-stage Build](http://blog.arungupta.me/smaller-java-image-docker-multi-stage-build/).
+
+   ```docker
+   # Use the official maven/Java 8 image to create a build artifact.
+   # https://hub.docker.com/_/maven
+   FROM maven:3.5-jdk-8-alpine as builder
+
+   # Copy local code to the container image.
+   WORKDIR /app
+   COPY pom.xml .
+   COPY src ./src
+
+   # Build a release artifact.
+   RUN mvn package -DskipTests
+
+   # Use AdoptOpenJDK for base image.
+   # It's important to use OpenJDK 8u191 or above that has container support enabled.
+   # https://hub.docker.com/r/adoptopenjdk/openjdk8
+   # https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
+   FROM adoptopenjdk/openjdk8:jdk8u202-b08-alpine-slim
+
+   # Copy the jar to the production image from the builder stage.
+   COPY --from=builder /app/target/helloworld-*.jar /helloworld.jar
+
+   # Run the web service on container startup.
+   CMD ["java", "-Djava.security.egd=file:/dev/./urandom", "-jar", "/helloworld.jar"]
+   ```
+Once you have recreated the sample code files (or used the files in the sample
```

Comment:
No need for this line, the title of the section already tells us we're building an app. Removed lines.